src/appleseed-max-impl/appleseedoslplugin/oslshaderregistry.h

```diff
@@ -46,18 +47,26 @@ class OSLShaderRegistry
     ClassDesc2* get_class_descriptor(int index) const;
     int get_size() const;
     void create_class_descriptors();
-    void add_parameter(
+    void add_const_parameter(
         ParamBlockDesc2*        pb_desc,
         const OSLParamInfo&     osl_param,
         MaxParam&               param_info,
         IdNameMap&              string_map,
+        const int               param_id,
+        int&                    ctrl_id,
```

Comment:
I find it a bit surprising that `add_const_parameter()` needs to modify `ctrl_id` and `string_id` but `add_input_parameter()` don't. Is there a fundamental reason why?

Reply:
Yes.. I've tried to make `add_parameter()` prettier by splitting in two functions and moving some increments outside.
Thing is that `add_const_parameter()` creates variable number of controls (like three point3 spinners) and string objects (for comboboxes).
`add_input_parameter()` add fixed number of controls, that's why it doesn't change ctrl_id from inside the function body.

Reply:
Ok, thanks for the explanation.